test/puppetlabs/puppetdb/cli/import_export_roundtrip_test.clj

```diff
@@ -82,86 +96,100 @@
                         :biz {:a [3.14 2.71] :b "the b" :c [1 2 3] :d {:e nil}}}
                :producer_timestamp (to-string (now))}
-        export-out-file (testutils/temp-file "export-test" ".tar.gz")
+        export-out-dir (testutils/temp-dir "export-test")
+        export-out-file-path (str (.getPath export-out-dir) "/outfile.tar.gz")
         catalog (-> (get-in wire-catalogs [6 :empty])
-                    (assoc :certname "foo.local"
+                    (assoc :certname certname
                            :producer_timestamp (now)))
-        report (:basic reports)]
+        report (-> (:basic reports)
+                   (assoc :certname certname))]
 
     (svc-utils/puppetdb-instance
-      (fn []
-        (is (empty? (export/get-nodes *base-url*)))
-
-        (svc-utils/sync-command-post (command-base-url *base-url*) "replace catalog" 6 catalog)
-        (svc-utils/sync-command-post (command-base-url *base-url*) "store report" 5
-                                     (tur/munge-example-report-for-storage report))
-        (svc-utils/sync-command-post (command-base-url *base-url*) "replace facts" 4 facts)
-
-        (is (testutils/=-after? munge-catalog catalog (->> (:certname catalog)
-                                                           (export/catalogs-for-node *base-url*)
-                                                           first)))
-
-        (is (testutils/=-after? munge-report report (->> (:certname report)
-                                                         (export/reports-for-node *base-url*)
-                                                         first)))
-        (is (= facts (->> (:certname facts)
-                          (export/facts-for-node *base-url*)
-                          first)))
-
-        (#'export/-main "--outfile" export-out-file
-                        "--host" (:host *base-url*)
-                        "--port" (str (:port *base-url*)))))
+     (fn []
+       (let [query-fn (partial query (tk-app/get-service svc-utils/*server* :PuppetDBServer))
+             submit-command-fn (partial submit-command (tk-app/get-service svc-utils/*server* :PuppetDBCommand))]
+         (is (empty? (query-fn :nodes helpers/api-version nil nil doall)))
+
+         (svc-utils/sync-command-post (command-base-url svc-utils/*base-url*) "replace catalog" 6 catalog)
+         (svc-utils/sync-command-post (command-base-url svc-utils/*base-url*) "store report" 5
+                                      (tur/munge-example-report-for-storage report))
+         (svc-utils/sync-command-post (command-base-url svc-utils/*base-url*) "replace facts" 4 facts)
+
+         (is (testutils/=-after? munge-catalog catalog (->> certname
+                                                            (export/catalogs-for-node (partial helpers/query-wrapper query-fn)
+                                                                                      (partial helpers/completion-wrapper query-fn))
```

Comment:
This is ugly, but like I said before, a refactor PR will follow this one up to clean this up.